Source/WebCore/platform/network/curl/CurlMultipartHandle.cpp

```diff
@@ -34,309 +34,260 @@
 #include "CurlResponse.h"
 #include "HTTPHeaderNames.h"
 #include "HTTPParsers.h"
+#include "ParsedContentType.h"
 #include "ResourceResponse.h"
 #include "SharedBuffer.h"
 #include <wtf/StringExtras.h>
 
 namespace WebCore {
 
-std::unique_ptr<CurlMultipartHandle> CurlMultipartHandle::createIfNeeded(CurlMultipartHandleClient& client, const CurlResponse& response)
+static std::optional<String> extractBoundary(const CurlResponse& response)
 {
-    auto boundary = extractBoundary(response);
-    if (!boundary)
-        return nullptr;
+    static const unsigned contentTypeLength = strlen("content-type:");
 
-    return makeUnique<CurlMultipartHandle>(client, *boundary);
-}
-
-std::optional<String> CurlMultipartHandle::extractBoundary(const CurlResponse& response)
-{
     for (auto header : response.headers) {
-        auto splitPosition = header.find(':');
-        if (splitPosition == notFound)
+        if (!header.startsWithIgnoringASCIICase("content-type:"_s))
             continue;
 
-        auto key = header.left(splitPosition).trim(deprecatedIsSpaceOrNewline);
-        if (!equalIgnoringASCIICase(key, "Content-Type"_s))
-            continue;
-
-        auto contentType = header.substring(splitPosition + 1).trim(deprecatedIsSpaceOrNewline);
-        auto mimeType = extractMIMETypeFromMediaType(contentType);
-        if (!equalLettersIgnoringASCIICase(mimeType, "multipart/x-mixed-replace"_s))
-            continue;
+        auto contentType = ParsedContentType::create(header.substring(contentTypeLength));
 
-        auto boundary = extractBoundaryFromContentType(contentType);
-        if (!boundary)
+        if (!equalLettersIgnoringASCIICase(contentType->mimeType(), "multipart/x-mixed-replace"_s))
             continue;
 
-        return String("--" + *boundary);
+        if (auto boundary = contentType->parameterValueForName("boundary"_s); !boundary.isEmpty())
+            return makeString("--", boundary);
     }
```

Comment:
Do you assume multiple content-type: headers?

Reply:
No, I don't. This code uses the first occurrence of "Content-Type: multipart/x-mixed-replace".

Reply:
Then, you should `return std::nullopt;` here if boundary parameter is not found.

Reply:
Fixed